src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,446 @@
+	}
+
+	private _supportsIndex: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Indexed Color",
+	})
+	public get supportsIndex(): boolean {
+		return this._supportsIndex;
+	}
+
+	public toJSON(): JSONObject {
+		return super.toJSONInherited({
+			supportsWarmWhite: this._supportsWarmWhite,
+			supportsColdWhite: this._supportsColdWhite,
+			supportsRed: this._supportsRed,
+			supportsGreen: this._supportsGreen,
+			supportsBlue: this._supportsBlue,
+			supportsAmber: this._supportsAmber,
+			supportsCyan: this._supportsCyan,
+			supportsPurple: this._supportsPurple,
+			supportsIndex: this._supportsIndex,
+		});
+	}
+}
+
+@CCCommand(ColorSwitchCommand.SupportedGet)
+@expectedCCResponse(ColorSwitchCCSupportedReport)
+export class ColorSwitchCCSupportedGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | CCCommandOptions,
+	) {
+		super(driver, options);
+	}
+}
+
+@CCCommand(ColorSwitchCommand.Report)
+export class ColorSwitchCCReport extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		// Docs say 'variable length', but the table shows 2 bytes.
+		validatePayload(this.payload.length >= 2);
+		this._colorComponent = this.payload[0];
+		this._value = this.payload[1];
+
+		// TODO: We probably want to store each color value
+		//	as a different ccValue.  How do we do that when
+		//	we get each color code separately in a single report?
+	}
+
+	private _colorComponent: ColorComponent;
+	public get colorComponent(): ColorComponent {
+		return this._colorComponent;
+	}
+
+	private _value: number;
+	public get value(): number {
+		return this._value;
+	}
+}
+
+interface ColorSwitchCCGetOptions extends CCCommandOptions {
+	colorComponent: ColorComponent;
+}
+
+@CCCommand(ColorSwitchCommand.Get)
+@expectedCCResponse(ColorSwitchCCReport)
+export class ColorSwitchCCGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | ColorSwitchCCGetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			// Populate properties from options object
+			this._colorComponent = options.colorComponent;
+		}
+	}
+
+	private _colorComponent: ColorComponent;
+	public get colorComponent(): ColorComponent {
+		return this._colorComponent;
+	}
+	public set colorComponent(value: ColorComponent) {
+		if (!ColorComponent[value]) {
+			throw new Error(
+				"colorComponent must be a valid color component index.",
+			);
+		}
+		this._colorComponent = value;
+	}
+
+	public serialize(): Buffer {
+		this.payload = Buffer.allocUnsafe(1);
+		this.payload[0] = this._colorComponent;
```

Comment:
For the sake of consistency, I would use `this.payload = Buffer.from([this._colorComponent])`.